mlir/include/mlir/Dialect/AMDGPU/IR/AMDGPU.td

```diff
@@ -126,11 +126,7 @@ def AMDGPU_RawBufferLoadOp :
                    DefaultValuedAttr<BoolAttr, "true">:$boundsCheck,
                    OptionalAttr<I32Attr>:$indexOffset,
                    Optional<I32>:$sgprOffset)>,
-    Results<(outs AnyTypeOf<[BF16, F16, F32, I32, I8, F8E5M2FNUZ, F8E4M3FNUZ,
-                              VectorOfLengthAndType<[2, 4], [F32, I32]>,
-                              VectorOfLengthAndType<[2, 4, 8], [F16, BF16]>,
-                              VectorOfLengthAndType<[2, 4, 8, 16],
-                                [I8, F8E5M2FNUZ, F8E4M3FNUZ]>]>:$value)> {
+    Results<(outs AnyType:$value)> {
```

Comment:
Isn't `AnyType` a bit too unrestrictive? Why not adding the "illegal" types you plan to support to the set of supported types?

Reply:
1. That's what `raw_buffer_store` already has
2. I'd rather not have to carry patches to add `i4`, `!my.custom.thing.that.lowers.to.iN`, and so on and so forth